Content.Server/Store/Systems/StoreSystem.Ui.cs

```diff
@@ -30,6 +36,16 @@ private void InitializeUi()
         SubscribeLocalEvent<StoreComponent, StoreRequestUpdateInterfaceMessage>(OnRequestUpdate);
         SubscribeLocalEvent<StoreComponent, StoreBuyListingMessage>(OnBuyRequest);
         SubscribeLocalEvent<StoreComponent, StoreRequestWithdrawMessage>(OnRequestWithdraw);
+        SubscribeLocalEvent<StoreComponent, StoreRequestRefundMessage>(OnRequestRefund);
+        SubscribeLocalEvent<StoreComponent, RefundEntityDeletedEvent>(OnRefundEntityDeleted);
+    }
+
+    private void OnRefundEntityDeleted(EntityUid uid, StoreComponent component, RefundEntityDeletedEvent args)
+    {
+        if (component.BoughtEntities.Contains(args.Uid))
+        {
+            component.BoughtEntities.Remove(args.Uid);
+        }
     }
```

Comment:
Remove doesn't fail if it doesn't exist so just call it directly (it returns a bool if it did exist).